common/log.go

```diff
@@ -0,0 +1,102 @@
+package common
+
+import (
+	"fmt"
+	"log"
+	"log/slog"
+	"os"
+	"strings"
+)
+
+type ILogger interface {
```

Comment:
Why do you need your own Logger? Can't you just use `log` or `slog`? If it's needed could you please add a comment explaining why you provide your own implementation?

Reply:
It's an abstraction to facilitate centralizing the logger init and hiding the impl.  This allows it to be swapped out in the future.  Passing around the logger seemed messy given the size of this app.